lib/mongo/collection/view/writable.rb

```diff
@@ -244,6 +253,19 @@ def update(spec, multi, opts)
             ).execute(server)
           end
         end
+
+        def apply_array_filters!(doc, server, opts = {})
+          if filters = opts[:array_filters] || opts[:arrayFilters] || opts['array_filters'] || opts['arrayFilters']
```

Comment:
You only need to check for ``'array_filters'`` and ``:array_filters``
and we should use a constant somewhere for the string